pubtools/pulplib/_impl/model/unit/modulemd.py

```diff
@@ -62,3 +64,32 @@ class ModulemdUnit(Unit):
 
     .. versionadded:: 2.6.0
     """
+    artifacts = pulp_attrib(
+        default=None,
+        type=list,
+        converter=frozenlist_or_none_converter,
+        pulp_field="artifacts",
+    )
+    """List of artifacts related to the given module, usually it's a list
+    rpms in nevra format without '.rpm' extension and consist of binary, debug and source
+    rpms as well.
+    """
+
+    profiles = pulp_attrib(type=dict, pulp_field="profiles", default=None)
+    """The profiles of this modulemd unit."""
+
+    def get_artifacts_filenames(self, exclude_srpms=False):
```

Comment:
A few suggestions for this one:

- I don't think it should be named with "get".  get prefix shouldn't be needed for a simple accessor I think.  In the existing cases in this library where "get_" is used, the operation involves some remote call to Pulp, which is different to this.
- I don't think the SRPM exclusion should be here. It means the method is doing two unrelated things ("calculate filenames" and "filter according to srpm or not"), and the specific combination is fairly arbitrary.  I guess you added exclude_srpms just because that was your current use-case, but other code might have other filtering needs like picking only certain arches.  Probably it should be left to the caller.
- Once exclude_srpms is removed, `@property` seems to fit well here

So for example, if it's made a `@property` named artifact_filenames then it would be something like this:

```python
# This gives me all the NEVRAs
artifacts = modulemd.artifacts

# If I want filenames instead I do this, which looks similar
rpms = modulemd.artifact_filenames

# And if I don't want SRPMs I can easily filter them myself
rpms = [r for r in rpms if not r.endswith('.src.rpm')]
```